go-controller/pkg/ovn/address_set/address_set_test.go

```diff
@@ -123,19 +145,32 @@ var _ = ginkgo.Describe("OVN Address Set operations", func() {
 					addr1 string = "1.2.3.4"
 					addr2 string = "5.6.7.8"
 				)
-
-				_, err := config.InitConfig(ctx, fexec, nil)
+				dbSetup := libovsdbtest.TestSetup{
+					NBData: []libovsdbtest.TestData{
+						&nbdb.AddressSet{
+							UUID:        "",
+							Name:        hashedAddressSet(addrsetName + ipv4AddressSetSuffix),
+							ExternalIDs: map[string]string{"name": addrsetName + ipv4AddressSetSuffix},
+							Addresses:   []string{"10.10.10.10"},
+						},
+					},
+				}
+				libovsdbOvnNBClient, _, err := libovsdbtest.NewNBSBTestHarness(dbSetup, stopChan)
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				asFactory = NewOvnAddressSetFactory(libovsdbOvnNBClient)
 
-				fexec.AddFakeCmd(&ovntest.ExpectedCmd{
-					Cmd:    "ovn-nbctl --timeout=15 --data=bare --no-heading --columns=_uuid find address_set name=a16990491322166530807",
-					Output: fakeUUID,
-				})
-				fexec.AddFakeCmdsNoOutputNoError([]string{
-					`ovn-nbctl --timeout=15 set address_set ` + fakeUUID + ` addresses="` + addr1 + `" "` + addr2 + `"`,
-				})
+				_, err = config.InitConfig(ctx, fexec, nil)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
 				_, err = asFactory.NewAddressSet("foobar", []net.IP{net.ParseIP(addr1), net.ParseIP(addr2)})
+				expectedDatabaseState := &nbdb.AddressSet{
+					Name:        hashedAddressSet(addrsetName + ipv4AddressSetSuffix),
+					Addresses:   []string{ipAddress1, ipAddress2},
+					ExternalIDs: map[string]string{"name": addrsetName + ipv4AddressSetSuffix},
+				}
+				gomega.Eventually(libovsdbOvnNBClient).Should(libovsdbtest.HaveTestDataIgnoringUUIDs(expectedDatabaseState))
+				addrSetList := &[]nbdb.AddressSet{}
+				libovsdbOvnNBClient.List(addrSetList)
 				gomega.Expect(err).NotTo(gomega.HaveOccurred())
 				gomega.Expect(fexec.CalledMatchesExpected()).To(gomega.BeTrue(), fexec.ErrorDesc)
```

Comment:
fexec expects could be removed? On this and other instances throughout the file.

Reply:
done